src/core/thread/mle.cpp

```diff
@@ -4743,5 +4762,48 @@ void Mle::DelayedResponseMetadata::RemoveFrom(Message &aMessage) const
     SuccessOrAssert(aMessage.SetLength(aMessage.GetLength() - sizeof(*this)));
 }
 
+Error Mle::DetachGracefully(otDetachGracefullyCallback aCallback, void *aContext)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(aCallback != nullptr, error = kErrorInvalidArgs);
+
+    if (IsDisabled() || IsDetached())
```

Comment:
Should `DetachGracefully` transfer state form Detached to Disabled and trigger the callback as a success?

Otherwise node would attach automatically.

Reply:
It's arguable that we can even trigger a success when `IsDisabled()` so that callers can always expect a success callback.  
It seems unnecessary for the caller to handle such a case differently.

Reply:
I agree that it should return a success when disabled. Same as `thread stop`.

Reply:
Done.